lib/Sema/CSDiagnostics.cpp

```diff
@@ -2943,6 +2943,12 @@ bool ContextualFailure::tryIntegerCastFixIts(
   // Add a wrapping integer cast.
   std::string convWrapBefore = toType.getString();
   convWrapBefore += "(";
+
+  // Add 'value' label for `NSNumber` to prevent roundabout error message
+  // See rdar://problem/82828226
+  if (convWrapBefore == "NSNumber(")
```

Comment:
Maybe would make sense to compare types instead :)
```suggestion
  auto &ctx = getASTContext();
  if (toType->isEqual(ctx.getNSNumberType())
```

Reply:
Yeah, checking the type itself is what we want here.

Reply:
Also I think we should add a check for NSInteger as well (at least)?

Reply:
Ah yeah, I had forgot about `NSInteger`. How many types integer can be bridged to? Because if there are more maybe we can check for `TypeChecker::isObjCBridgedTo` instead?

Reply:
At least `NSUInteger` and `NSDecimal` I know of. `isObjCBridgedTo` indeed sounds like a better check if we were to suggest a coercion here instead of trying to enumerate the types.

Reply:
Yeah, and if types are not bridgeable we can just fallback to the current fix-it instead

Reply:
Sounds good!